src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/ogcapi/APIFilterParser.java

```diff
@@ -75,9 +91,21 @@ public Filter parse(String filter, String filterLang) {
             // in OGC APIs assume CRS84 as the default, but the underlying machinery may default to
             // the native CRS in EPSG axis order instead, best making the CRS explicit instead
             if (parsedFilter != null) {
-                DefaultCRSFilterVisitor crsDefaulter =
-                        new DefaultCRSFilterVisitor(FF, DefaultGeographicCRS.WGS84);
-                return (Filter) parsedFilter.accept(crsDefaulter, null);
+                DefaultCRSFilterVisitor crsDefaulter = new DefaultCRSFilterVisitor(FF, filterCRS);
+                parsedFilter = (Filter) parsedFilter.accept(crsDefaulter, null);
+
+                // if there is not FeatureType provided, we can not reproject
+                if (ft != null) {
```

Comment:
I believe this is actually redundant, every feature source returned by GeoServer is wrapped by GeoServerFeatureSource which, among other things, takes care of reprojecting to native:
https://github.com/geoserver/geoserver/blob/9e40cbbccb9626b8d08516c8eacaf5a8f168189b/src/main/src/main/java/org/vfny/geoserver/global/GeoServerFeatureSource.java#L398